webapi/src/main/scala/org/knora/webapi/responders/v2/ValuesResponderV2.scala

```diff
@@ -1174,20 +1162,24 @@ final case class ValuesResponderV2Live(
                  for {
                    checkNode <- resourceUtilV2.checkListNodeExistsAndIsRootNode(listValue.valueHasListNode)
 
-                   _ = checkNode match {
-                         // it doesn't have isRootNode property - it's a child node
-                         case Right(false) => ()
-                         // it does have isRootNode property - it's a root node
-                         case Right(true) =>
-                           throw BadRequestException(
-                             s"<${listValue.valueHasListNode}> is a root node. Root nodes cannot be set as values."
-                           )
-                         // it deosn't exists or isn't valid list
-                         case Left(_) =>
-                           throw NotFoundException(
-                             s"<${listValue.valueHasListNode}> does not exist or is not a ListNode."
-                           )
-                       }
+                   _ <- checkNode match {
+                          // it doesn't have isRootNode property - it's a child node
+                          case Right(false) => ZIO.unit
+                          // it does have isRootNode property - it's a root node
+                          case Right(true) =>
+                            ZIO.fail(
+                              BadRequestException(
+                                s"<${listValue.valueHasListNode}> is a root node. Root nodes cannot be set as values."
+                              )
+                            )
+                          // it doesn't exists or isn't valid list
+                          case Left(_) =>
+                            ZIO.fail(
+                              NotFoundException(
+                                s"<${listValue.valueHasListNode}> does not exist or is not a ListNode."
+                              )
+                            )
+                        }
```

Comment:
The entire block starting from the line 1159 is the same as from lines 224-252. Does it make sense to extract it separate method?

Reply:
Yes, you are right. I have extracted the duplicate code into its own method:
https://github.com/dasch-swiss/dsp-api/pull/2799/commits/111d13f5c755fbf99fd5d7bb5ddab9540cadd6ee